main.go

```diff
@@ -0,0 +1,28 @@
+package main
```

Comment:
> Yup. From what's here, it seems like server/message.go and server/serialization.go can move to protocol/ since I imagine they'd be used by both client/ and server/. But correct me if I'm wrong.

It makes sense. 

> But how I originally thought about these was that they'd be separate processes, and the server would only accept registrations from the privileged places (ie. only registrations from localhost)

Yeah, same thought here. 

For now, I would go with your advices. Thanks for clarifying!